lighthouse-core/lib/lh-error.js

```diff
@@ -24,7 +59,7 @@ class LighthouseError extends Error {
     super(errorDefinition.code);
     this.name = 'LHError';
     this.code = errorDefinition.code;
-    this.friendlyMessage = errorDefinition.message;
+    this.friendlyMessage = str_(errorDefinition.message, errorDefinition);
```

Comment:
What if the second argument to the `LHError` constructor becomes the replacement values? I feel like it would be conceptually easier to understand without having all of it on the first argument and

```js
const err = new LHError(
  LHError.errors.PROTOCOL_TIMEOUT,
  protocolMethod: `${method}`,
);
reject(err);
```

is a lot easier to understand at a glance than

```js
const err = new LHError({
  ...LHError.errors.PROTOCOL_TIMEOUT,
  protocolMethod: `${method}`,
});
reject(err);
```

I was going to say ideally we'd do this but the existing `properties` second arg makes that a problem, but I think it's actually fine. The only place we ever use that argument is in this file for `fromProtocolMessage()`, and we use it for `protocolMethod` and `protocolError`, one of which is already an icu argument name :)

What if `properties` just did both? It puts its properties on the error object (as it does now), and it gets passed into `str_` (as `errorDefinition` is doing here).

Reply:
I think later on we should do stricter type checking...if you use a string that needs a replacement, you should have to provide the replacement